rllib/agents/trainer.py

```diff
@@ -483,6 +483,8 @@
     #  - a path/URI to save to a custom output directory (e.g., "s3://bucket/")
     #  - a function that returns a rllib.offline.OutputWriter
     "output": None,
+    # Arguments accessible from the IOContext for configuring custom output
```

Comment:
Very cool. Now we have this concept (defining `type`, or callable returning instance, or (str) classpath) for OutputWriter, Exploration, ReplayBuffer!
We should unify this way of configuring stuff more - in a separate PR, of course. :)

Reply:
Totally, let's discuss more. I am not clear how Exploration fits in this framework.
We should also unify InputReader with ReplayBuffer input I feel like.